src/autopas/containers/octree/OctreeLeafNode.h

```diff
@@ -173,10 +175,27 @@ class OctreeLeafNode : public OctreeNodeInterface<Particle>, public FullParticle
    */
   void markAlreadyProcessed(OctreeLeafNode<Particle> *other) { _alreadyProcessed.push_back(other); }
 
+  /**
+   * Get the assigned id of this leaf node
+   * @return An ID (or -1 if there was no ID assigned to this node)
+   */
+  int getID() { return _id; }
+
+  /**
+   * Set the ID of this node
+   * @param id An integer ID
+   */
+  void setID(int id) { this->_id = id; }
+
  private:
   /**
    * The list that contains the neighbors that have already been processed in one traversal run.
    */
   std::vector<OctreeLeafNode<Particle> *> _alreadyProcessed;
+
+  /**
+   * The ID assigned to this node (-1 if unassigned)
+   */
+  int _id;
```

Comment:
```suggestion
  int _id{-1};
```
and remove the assignment in the constructor